File: app/src/main/kotlin/me/tomasan7/jecnadesktop/ui/scene/MainScene.java
package me.tomasan7.jecnadesktop.ui.scene;

import me.tomasan7.jecnadesktop.JecnaDesktop;
import me.tomasan7.jecnadesktop.ui.FXMLPage;
import me.tomasan7.jecnadesktop.ui.JDScene;
import me.tomasan7.jecnadesktop.ui.controller.LoginPageController;
import me.tomasan7.jecnadesktop.ui.controller.MainPageController;

public class MainScene extends FXMLPage
{
	private final JecnaDesktop jecnaDesktop;

	public MainScene (JecnaDesktop jecnaDesktop)
	{
		this.jecnaDesktop = jecnaDesktop;
	}

	@Override
	public String getFxmlLocation ()
	{
		return JDScene.MAIN.getLocation();
	}

	@Override
	public Object getController ()
	{
		return new MainPageController(jecnaDesktop);
	}
}

File: app/src/main/kotlin/me/tomasan7/jecnadesktop/ui/scene/MainScene.kt
package me.tomasan7.jecnadesktop.ui.scene

import me.tomasan7.jecnadesktop.JecnaDesktop
import me.tomasan7.jecnadesktop.ui.FxmlPage
import me.tomasan7.jecnadesktop.ui.JDScene
import me.tomasan7.jecnadesktop.ui.controller.MainPageController

class MainScene(jecnaDesktop: JecnaDesktop) : FxmlPage()
{
    override val fxmlLocation = JDScene.MAIN.location
    override val controller =  MainPageController(jecnaDesktop)
}